src/Packlist/PacklistForm.js

```diff
@@ -0,0 +1,218 @@
+          <PacklistNameInputLabel htmlFor="name">
+            Name of the new PackList:
+          </PacklistNameInputLabel>
+          <Input
+            placeholder="PackList name"
+            id="name"
+            name="name"
+            ref={register({
+              required: true,
+              minLength: 3,
+              maxLength: 10,
+              validate: {
+                length: (value) =>
+                  value?.trim().length >= 3 && value?.trim().length <= 10,
+                nameTaken: (value) =>
+                  !packlists.find((packlist) => packlist.name === value),
+              },
+            })}
+          />
+          {errors.name?.type === 'required' && (
+            <ErrorMessage>Name is required!</ErrorMessage>
+          )}
+          {errors.name?.type === 'nameTaken' && (
+            <ErrorMessage>Name is taken!</ErrorMessage>
+          )}
+          {(errors.name?.type === 'validate' ||
+            errors.name?.type === 'minLength') && (
+            <ErrorMessage>
+              This field requires at least 3 characters!
+            </ErrorMessage>
+          )}
+          {(errors.name?.type === 'validate' ||
+            errors.name?.type === 'maxLength') && (
+            <ErrorMessage>
+              The name can reach a maximum of 10 characters!
+            </ErrorMessage>
+          )}
+          <Label htmlFor="itemInput">Create new item or task:</Label>
+          <Input
+            placeholder="item you need or task you have to do"
+            id="itemInput"
+            name="item"
+            onKeyDown={handlePacklistKeyDown}
+            ref={(el) => {
+              itemRef.current = el
+              // register(el, {
+              //   required: true,
+              //   minLength: 3,
+              //   maxLength: 20,
+              //   validate: (value) =>
+              //     value &&
+              //     value.trim().length >= 3 &&
+              //     value.trim().length <= 20,
+              // })
+            }}
+          />
+          <HiddenInput
+            name="atLeastOneItem"
+            ref={register({
+              validate: () => {
+                console.log('items', items)
+                return items.length > 0
+              },
+            })}
+          />
+
+          {errors.atLeastOneItem?.type === 'validate' && (
+            <ErrorMessage>We need one item!</ErrorMessage>
+          )}
+          {(errors.item?.type === 'validate' ||
+            errors.item?.type === 'minLength') && (
+            <ErrorMessage>
+              This field requires at least 3 characters!
+            </ErrorMessage>
+          )}
+          {(errors.item?.type === 'validate' ||
+            errors.item?.type === 'maxLength') && (
+            <ErrorMessage>
+              The Item can reach a maximum of 20 characters!
+            </ErrorMessage>
+          )}
+
+          <AddButton
+            type="button"
+            onClick={(packlistitem) => {
```

Comment:
packlistitem is the first name which is not good recognizable. Maybe use of camleCase could be helpful

Reply:
yes, will be changed